### PR TITLE
Add file via upload

### PR DESCRIPTION
**Pull request by irwinj22**
Python code to filter through tag movements.

### CODE REVIEW

File: tag_filter.py [hunk 1]
```diff
@@ -0,0 +1,182 @@
...
+        self.real_name = self.tag_uuid_convert(value['tagUuid'], self.tag_name_data)
+        self.csv_data[self.count].append(self.real_name)
+        lat_1 = round(value['gpsLocation']['lat'], 2)
+        address = self.location_data_print(self.location_data, lat_1)
+        self.csv_data[self.count].append(address)
+        self.csv_data[self.count].append(value['movement'])
+        self.csv_data[self.count].append(self.human_time(value['timestampMs']))
+        self.csv_data[self.count].append(self.count + 1)
+   
+        with open(self.args.csv + '.csv', 'w', newline = '', encoding='UTF8') as f:
```
Comment (rhombusdev) [inline tag_filter.py]: You should check if the provided csv filename already has the `.csv` extension.  I think that most people will provide it if they specify this parameter.

File: tag_filter.py [hunk 2]
```diff
@@ -0,0 +1,182 @@
...
+    def csv_add(self, value):
+        self.csv_data.append([])
+        self.real_name = self.tag_uuid_convert(value['tagUuid'], self.tag_name_data)
+        self.csv_data[self.count].append(self.real_name)
+        lat_1 = round(value['gpsLocation']['lat'], 2)
+        address = self.location_data_print(self.location_data, lat_1)
+        self.csv_data[self.count].append(address)
+        self.csv_data[self.count].append(value['movement'])
+        self.csv_data[self.count].append(self.human_time(value['timestampMs']))
+        self.csv_data[self.count].append(self.count + 1)
```
Comment (rhombusdev) [inline tag_filter.py]: Unsure what this column does - it is not tied to anything else and is unnecessary?

File: tag_filter.py (hunk 1 above)
Comment (rhombusdev) [inline tag_filter.py]: You are overwriting this file every time a new row is added.  You should open the csv file once for the entire program and then write each row to it.

File: tag_filter.py [hunk 3]
```diff
@@ -0,0 +1,182 @@
...
+    def execute(self):
+        self.header = ['Tag Name', 'Address', 'Movement', 'Date', 'Movement']
+        self.csv_data =[]
+        self.data = self.tag_data()  
+        self.tag_name_data = self.tag_name()    
+        self.location_data = self.locations()
+        self.count = 0
+        self.final_list = []
+
+        for event in self.data['locomotionEvents']:
```
Comment (rhombusdev) [inline tag_filter.py]: I would add some more logging around what happens if someone provides the wrong name of a tag or specifies a time period in which there are no locomotion events.  Right now I think the program exits and no csv file is written if either of the above occurs.

**Review by rhombusdev — COMMENTED** [review]
Some cleanup and fixes.

**Comment by irwinj22** [comment]
I just added another file that hopefully addressed the suggestions that were made.

**Comment by rhombusdev** [comment]
We already have a get_frame example that does what your screenshot example does.  No need to include screenshot.py in this pull request.

**Pull request closed without merging**